modules/generator/config.go

```diff
@@ -98,9 +115,19 @@ func (cfg *Config) Validate() error {
 		}
 	}
 
+	if !slices.Contains(validCodecs, cfg.Codec) {
+		return fmt.Errorf("invalid codec: %s, valid choices are %s", cfg.Codec, validCodecs)
+	}
+
 	return nil
 }
 
+func (cfg *Config) shouldJoinRing() bool {
+	// If both gRPC and the localblocks processor are disabled, generators don't need
+	// to join the ring.
+	return !(cfg.DisableLocalBlocks && cfg.DisableGRPC)
```

Comment:
what does disabling local blocks have to do with joining the ring or not? are these not independent? if DisableGRPC=true it makes sense to me to not join the ring and skip all that logic, but I don't see the connection with disabling local blocks.

Reply:
I believe you're right and these two should be independent, I have changed it in https://github.com/grafana/tempo/pull/4686/commits/846e38008e6f2dd09653a5d3a0688dd325410c62.
My initial thinking was that localblocks' function was to produce data that can be queried via gRPC, but I think that understanding is incorrect.